### ISSUE TITLE
Mobile UI

### ISSUE THREAD

**Issue by lawl** [postmarketOS/pmbootstrap#62]
Weston is a cool demo, but far from a usable day-to-day shell people can work with.

We need to provide a sane UI.

Example code/projects:

* https://plasma-mobile.org/

Please contribute any other code/project you know of.

**Comment by ollieparanoid** [comment]
I personally think plasma mobile is the most complete one, put people should package what *they* like. More links:
* [ubports](https://ubports.com/) (Fork of the discontinued Ubuntu Phone project)
* [libweston](https://github.com/wayland-project/weston) to write your own minimal UI and get the device support from weston for free
* mer/SailfishOS related:
  * [lipstick](https://git.merproject.org/mer-core/lipstick) ([relevant comment](https://www.reddit.com/r/linux/comments/6djq1z/postmarketos_aiming_for_a_10_year_lifecycle_for/di3yc7w/))
  * [glacier](https://github.com/qwazix/glacier-controls-spec) ([relevant comment](https://www.reddit.com/r/linux/comments/6djq1z/postmarketos_aiming_for_a_10_year_lifecycle_for/di4iekw/))

**Comment by MartijnBraam** [comment]
I was definitively planning to mess around with libweston to make my own shell that mimics maemo

**Comment by vmx** [comment]
There's also [Gaia](https://github.com/mozilla-b2g/gaia) the UI from FirefoxOS/Boot2Gecko.

**Comment by dzervas** [comment]
There's enlightenment too (it's responsive)

-----Original Message-----

**Comment by jsprada** [comment]
AsteroidOS is a smart watch GNU/Linux OS that has a QT GUI.   Although it's
a watch, I think it can be used as a resource to consider as we develop a
sane "desktop" environment for phones.

https://asteroidos.org/


On Jun 5, 2017 9:21 AM, "Volker Mische" <notifications@github.com> wrote:

There's also Gaia <https://github.com/mozilla-b2g/gaia> the UI from
FirefoxOS/Boot2Gecko.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/postmarketOS/pmbootstrap/issues/62#issuecomment-306230853>,
or mute the thread

**Comment by kskarthik** [comment]
Enlightment sound good :)

**Comment by ollieparanoid** [comment]
Gnome 3 is in Alpine Linux testing now. I think it does not have Wayland support yet (because of the way GTK3 is packaged in Alpine), but if someone wants to, it shouldn't be that hard to change and try out. It isn't really a phone UI, but was made with touch devices in mind, has an on screen keyboard etc.

**Comment by JBBgameich** [comment]
Plasma Mobile already has packaging for Arch Linux. I think this is similar to what Alpine uses, so maybe you can reuse the existing work at https://github.com/blue-systems/pm-arch/tree/master/packages.

**Comment by Immortalin** [comment]
There is also Dart

**Comment by letoram** [comment]
Pitch: my project ( https://github.com/letoram/arcan/wiki ) has been in use for these kinds of things for a few years (point-of-sale, home automation / surveillance) and is designed specifically for being able to quickly put together interactive presentations and user-interfaces, from small embedded to full desktop with everything in place for prototyping on desktop and running on target without many surprises.

Bringup: Low on dependencies, works with musl-libc (use it for my desktops/laptops/tablets on alpine/void myself). It can run on the recent open source linux graphics stack, and on some of the binary blobs with tuning (I've personally used the egl-gles platform has been used for raspberry pi and the jolla phone). There's no software path to the graphics platform, though one can be hacked together rather quickly if you don't need 2d effects, but software raster will kill your memory bandwidth anyhow. Lots more when it comes to other controls (backlights, sensors, LEDs) but poke me on irc if you're curious, I'm in the channel.

I used to work as a senior systems engineer specialising in security and graphics at Sony Mobile (from pre-android feature phone days up to/including Lollipop) so most of the pitfalls I know of have all been accounted for ;-)

**Comment by AndrewMcSwain** [comment]
@letoram nice work, looks very thorough!

@Immortalin I agree with you that we should have a skin or interface that mimics the most modern android UI at first, with the added bonus of more configurability and "suggestions" that can guide the user into becoming more familiar with the freedom provide by a free and open source mobile OS.

This way we remain inclusive and don't have to re-do / throw out the work of UX and interface designers ( I'm sure they'd appreciate that :D ).

UX is all about expectation, and expectations with respect to mobile devices are A) highly personally B) deeply ingrained. Building new expectations takes time, but it is a very fruitful endeavor!

**Comment by pavelmachek** [comment]
I am running ofono stack on N900, and so I needed GUI to dial numbers, write&receive SMSes, display signal strength and battery, connect GPRS and something to ring when there's an incoming call. There was nothing for ofono I could see, so I wrote https://gitlab.com/tui/tui/blob/master/ofone/ofone . Now, python is a bit too heavy for this, so I started doing vala version (vfone, in same repository).

**Comment by ollieparanoid** [comment]
@pavelmachek in case you're interested, someone packaged hildon the other day for postmarketOS, and showed a screenshot running it in qemu (sorry, I don't recall the username, but great work!). But it's not in this repository yet (I hope we get a PR at some point), it's in one of the forks. The branch was called `maemo` I think.

**Comment by pavelmachek** [comment]
And I guess I should mention... I'm using Xfce on N900. It is not really finger-optimized, but is mostly usable with the stylus. I like it better than original Maemo. I also had MATE (Gnome 2) running on the box.

**Comment by zhuowei** [comment]
Tizen's UI is based on Enlightenment and is (mostly?) open source. E.g: the homescreen app is https://review.tizen.org/git/?p=profile/mobile/apps/native/homescreen-efl.git;a=tree;h=refs/heads/tizen_4.0;hb=refs/heads/tizen_4.0 . This may be another option.

**Comment by ollieparanoid** [comment]
PDA interfaces have been mentioned in /r/postmarketOS:
https://www.reddit.com/r/postmarketOS/comments/7582fb/long_term_interest_pda_interfaces/

They seem to have really outdated codebases though.

**Comment by pavelmachek** [comment]
I have used both Opie and GPE before.

Opie is simpler, QT-based, GPE is slightly newer, GTK-based and uses
real X system IIRC.

But they are meant to be operated using stylus, not finger, which is
really quite major difference. I'd not Opie to be too useful for
us. Parts of GPE (GPE-Contacts, GPE-Calendar, GPE-Games, GPE-ToDo)
might be useful if it is easy to get them running.

-- 
(english) http://www.livejournal.com/~pavelmachek
(cesky, pictures) http://atrey.karlin.mff.cuni.cz/~pavel/picture/horses/blog.html

**Comment by JLIT0** [comment]
@pavelmachek You're right.  But two devices which are being ported over right now, the Nokia N900 and the Galaxy Note II do support stylus input.

**Comment by craftyguy** [comment]
I don't think GPE has been touched in years: https://en.wikipedia.org/wiki/GPE_Palmtop_Environment


Opie seems to be slightly newer in terms of date of last release: https://en.wikipedia.org/wiki/Open_Palmtop_Integrated_Environment

They actually don't look _that_ bad, my main concern would be how nicely they play with musl and GCC6...

**Comment by ollieparanoid** [comment]
* "feature phone idea": #944 
* Symbian was OSS for a short time (2010-11): [source dumps](https://en.wikipedia.org/wiki/Symbian#Symbian.5E3_EPL_Source)

**Comment by pavelmachek** [comment]
Well, I'm pretty sure there were OSS stacks for OpenMoko phones. But
they are not based on ofono, and are probably obsolete. There's
QTembedded. But probably still better start than Symbian :-).

**Comment by Imerion** [comment]
Wow, getting OPIE to run on a modern (stylus-equipped) phone would be a dream come true for me! I remember how awesome it was back when I ran it on an old PDA. I could actually get lots of real work done quickly on that UI, something that can't be said of todays mobile UIs that seem to be designed solely for the purpose of passively consuming endlessly scrolling social media...

**Comment by PabloCastellano** [comment]
JFTR: The MATE Desktop Environment [has been merged](https://github.com/postmarketOS/pmbootstrap/pull/1012)!

**Comment by ollieparanoid** [comment]
@MartijnBraam and @rrcha talked about Aura from Google's chromium OS as possible UI.

**Comment by PureTryOut** [comment]
A better link for the Glacier UI source: https://github.com/nemomobile-ux/

**Comment by JBBgameich** [comment]
As there is now a PR for packaging i3 in postmarketOS, I remembered a project that configures i3 to be usable with touch screens and the n900. https://github.com/robotanarchy/penguinphone

**Comment by JBBgameich** [comment]
As there is now a PR for packaging i3 in postmarketOS, I remembered a project that configures i3 to be usable with touch screens and the n900. https://github.com/robotanarchy/penguinphone
